core/templates/pages/skill-editor-page/editor-tab/skill-editor-main-tab.directive.ts

```diff
@@ -83,6 +83,10 @@ angular.module('oppia').directive('skillEditorMainTab', [
             return $scope.subtopicName;
           };
 
+          $scope.updateViewPublishButton = function() {
+            $scope.$apply();
+          };
```

Comment:
Couple of questions:
1. Why are we triggering a digest loop here and will we get rid of this after migrating this directive?
2. The function name is a bit confusing, I can't find a "publish button" in the skill-editor-main-tab directive.

Reply:
1. we are going to remove this function when we are going to upgrade this file to angular and used `$scope.$apply();` because without this it is not detecting new changes made in `prerequisite-skills-editor`.
2.  code of publish button  is on the base file that's why you are unable to see that code. for now, I have renamed the function.

Take a look at #14501 expected changes video and proof video of this file you will understand the need for this function.

Reply:
Thanks! Hm, since the update doesn't actually correspond to the main editor tab directive, can we instead emit an event in the SkillUpdateService (or the SkillEditorStateService) and have that be subscribed in the nav-bar directive so that it updates the button status correctly? That way we don't need the extra component input/outputs. I noticed that we already have something similar:

https://github.com/oppia/oppia/blob/53667f37cce8b3522a1e1af28ff32aaa5de426f5/core/templates/pages/skill-editor-page/navbar/skill-editor-navbar.directive.ts#L147-L151

Reply:
using subscribe is only working when we are publishing changes otherwise it is not updating.
proof :->

https://user-images.githubusercontent.com/77006897/147820755-7e17731d-e34c-42fb-ac3a-946276d2b955.mp4

I Think using `scope.$apply()` is good in this case

https://user-images.githubusercontent.com/77006897/147821036-f1234fff-5d1f-40d3-98be-6a84fa4d15ec.mp4

@kevintab95 PTAL

Reply:
@kevintab95 PTAL resolved

Reply:
Thanks for addressing it. Just one minor change, instead of having the "main editor" update, can we more the "subscribe" block into `skill-editor-navbar` directive since the "publish button" (what needs to update) is actually present there?

Reply:
Yes i will do that

Reply:
@kevintab95 PTAL Done